lib/cfd_trading/cfd_offer.dart

```diff
@@ -86,96 +89,150 @@ class _CfdOfferState extends State<CfdOffer> {
+                const SizedBox(width: 2),
+                Text(fmtAsk, style: const TextStyle(fontWeight: FontWeight.w600))
+              ],
+            ),
+            Row(
+              children: [
+                const Text("index:"),
+                const SizedBox(width: 2),
+                Text(fmtIndex, style: const TextStyle(fontWeight: FontWeight.w600))
+              ],
+            ),
+          ],
         ),
-        Dropdown(
-            values: [ContractSymbol.BtcUsd.name.toUpperCase()],
-            onChange: (contractSymbol) {
-              order.contractSymbol =
-                  ContractSymbol.values.firstWhere((e) => e.name == contractSymbol!.toLowerCase());
-            },
-            value: order.contractSymbol.name.toUpperCase()),
-        Dropdown(
-            values: CfdOffer.leverages.map((l) => 'x$l').toList(),
-            onChange: (leverage) {
+        const SizedBox(height: 10),
+        PositionSelection(
+            onChange: (position) {
               setState(() {
-                order.leverage = int.parse(leverage!.substring(1));
+                order.position = position!;
+                order.openPrice = Position.Long == position ? offer.ask : offer.bid;
               });
             },
-            value: 'x' + order.leverage.toString()),
-      ]),
-      Column(mainAxisAlignment: MainAxisAlignment.start, children: <Widget>[
-        Container(
-          margin: const EdgeInsets.only(top: 25),
-          child: TtoTable([
-            TtoRow(
-                label: 'Liquidation Price',
-                value: liquidationPrice.toString(),
-                type: ValueType.usd),
-            TtoRow(label: 'Margin', value: margin, type: ValueType.satoshi),
-            TtoRow(label: 'Expiry', value: expiry, type: ValueType.date),
-          ]),
-        )
-      ]),
+            value: order.position),
+        Padding(
+          padding: const EdgeInsets.only(top: 20.0, bottom: 5.0),
+          // Add box constraints to make sure that the row does not push the elements below down when the quantity text field displays a validation error
+          child: ConstrainedBox(
+            constraints: const BoxConstraints(minHeight: 70),
+            child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                crossAxisAlignment: CrossAxisAlignment.baseline,
+                textBaseline: TextBaseline.alphabetic,
+                children: [
+                  Expanded(
+                    child: TextFormField(
+                      decoration: InputDecoration(
+                        contentPadding: const EdgeInsets.only(bottom: 0.0),
+                        alignLabelWithHint: true,
+                        enabledBorder: UnderlineInputBorder(
+                            borderSide: BorderSide(
+                                color: Theme.of(context).colorScheme.primary, width: 2.0)),
+                        labelText: "* Quantity (10..1000)",
```

Comment:
🙃 I would omit the range. Maybe it can be displayed below the text field when the user has it selected?

Reply:
> Maybe it can be displayed below the text field when the user has it selected?

That's tricky, because the state of the `FormTextField` would become quite complex. It then has to decide if the error is to replace the (second) hint. It certainly does not do that out of the box :)

Reply:
Haha I thought so.

Maybe the range is fine, let's see what the others think.

Reply:
The notion `(10..1000)` might not be understandable for normal users, what do you think of changing it to `Min:10, Max: 1000`?

Second thought: do we need this label? If we have the one below, that should be good enough I guess?

Reply:
I think you are right with your second thought. It's probably enough if the user gets told the min and max when they input something invalid.